src/Discord.Net.WebSocket/BaseSocketClient.Events.cs

```diff
@@ -165,6 +165,13 @@ public partial class BaseSocketClient
             remove { _userVoiceStateUpdatedEvent.Remove(value); }
         }
         internal readonly AsyncEvent<Func<SocketUser, SocketVoiceState, SocketVoiceState, Task>> _userVoiceStateUpdatedEvent = new AsyncEvent<Func<SocketUser, SocketVoiceState, SocketVoiceState, Task>>();
+        /// <summary> Fired when the bot connects/disconnects to a Discord voice server. </summary>
```

Comment:
According to [Discord's docs](https://discordapp.com/developers/docs/topics/gateway#voice-server-update), this event fires only on connecting, not disconnecting, as well as when the current voice instances changes servers.